File: system-test/dotnet/SystemTests/Latest/ExternalSystemContractTests/Clock/ClockStubContractTest.cs
using Dsl.Port;

namespace SystemTests.Latest.ExternalSystemContractTests.Clock;

public class ClockStubContractTest : BaseClockContractTest
{
    protected override ExternalSystemMode? FixedExternalSystemMode => ExternalSystemMode.Stub;

    [Fact]
    public async Task ShouldBeAbleToGetConfiguredTime()
    {
        (await Scenario()
            .Given().Clock().WithTime("2024-01-02T09:00:00Z")
            .Then().Clock())
            .HasTime("2024-01-02T09:00:00Z");
    }
}

File: system-test/java/src/test/java/com/optivem/shop/systemtest/latest/contract/clock/ClockStubContractTest.java
package com.optivem.shop.systemtest.latest.contract.clock;

import com.optivem.shop.systemtest.configuration.ExternalSystemMode;
import com.optivem.testing.Isolated;
import org.junit.jupiter.api.Test;

@Isolated
class ClockStubContractTest extends BaseClockContractTest {
    @Override
    protected ExternalSystemMode getFixedExternalSystemMode() {
        return ExternalSystemMode.STUB;
    }

    @Test
    void shouldBeAbleToGetConfiguredTime() {
        scenario
                .given().clock().withTime("2024-01-02T09:00:00Z")
                .then().clock().hasTime("2024-01-02T09:00:00Z");
    }
}
